### ISSUE TITLE
[KNOWN ISSUE] python libraries aren't being installed properly

### ISSUE THREAD

**Issue by Deathdadev** [pinokiofactory/TRELLIS#1]
**The current state of this repository is incomplete and still under active development.**

### Root Cause of the Error
The issue arises because the project requirements have been directly copied from the [TRELLIS-for-Windows repository](https://github.com/sdbds/TRELLIS-for-windows). Additionally, these requirements are hard-coded into the install.js script, without proper translation or adaptation to fit the Pinokio project.

### Proposed Solutions

- Properly adapt and validate the requirements for Pinokio.
- Refactor the install.js script to dynamically fetch and configure dependencies specific to this project.
- **Remove requirements.txt from repository, and properly handle installing python dependencies using the pinokio scripts**

**Comment by Deathdadev** [comment]
I am feeling real lazy, so I am going to use aider to convert the original https://raw.githubusercontent.com/microsoft/TRELLIS/refs/heads/main/setup.sh to a pinokio script

**Comment by Deathdadev** [comment]
### Recent changes to the installation logic attempted to mimic the original installation scripts, adapting them to work across user systems.

### Observed Issue
During installation, the following error occurs:

```
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Collecting git+https://github.com/JeffreyXiang/diffoctreerast.git
  Cloning https://github.com/JeffreyXiang/diffoctreerast.git to c:\pinokio\cache\tmpdir\pip-req-build-bp3x90gm
  Running command git clone --filter=blob:none --quiet https://github.com/JeffreyXiang/diffoctreerast.git 'C:\pinokio\cache\TMPDIR\pip-req-build-bp3x90gm'
  Resolved https://github.com/JeffreyXiang/diffoctreerast.git to commit b09c20b84ec3aace4729e6e18a613112320eca3a
  Running command git submodule update --init --recursive -q
  Preparing metadata (setup.py) ... done
Installing collected packages: diffoctreerast
  DEPRECATION: diffoctreerast is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for diffoctreerast ... error
  error: subprocess-exited-with-error

  × Running setup.py install for diffoctreerast did not run successfully.
  │ exit code: 1
  ╰─> [43 lines of output]
      running install
      C:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-310
      creating build\lib.win-amd64-cpython-310\diffoctreerast
      copying diffoctreerast\octree_decoupoly_rasterizer.py -> build\lib.win-amd64-cpython-310\diffoctreerast
      copying diffoctreerast\octree_gaussian_rasterizer.py -> build\lib.win-amd64-cpython-310\diffoctreerast
      copying diffoctreerast\octree_trivec_rasterizer.py -> build\lib.win-amd64-cpython-310\diffoctreerast
      copying diffoctreerast\octree_voxel_rasterizer.py -> build\lib.win-amd64-cpython-310\diffoctreerast
      copying diffoctreerast\__init__.py -> build\lib.win-amd64-cpython-310\diffoctreerast
      running build_ext
      C:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\utils\cpp_extension.py:497: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
        warnings.warn(msg.format('we could not find ninja.'))
      C:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\utils\cpp_extension.py:382: UserWarning: Error checking compiler version for cl: [WinError 2] The system cannot find the file specified
        warnings.warn(f'Error checking compiler version for {compiler}: {error}')
      C:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\utils\cpp_extension.py:416: UserWarning: The detected CUDA version (12.1) has a minor version mismatch with the version that was used to compile PyTorch (12.4). Most likely this shouldn't be a problem.
        warnings.warn(CUDA_MISMATCH_WARN.format(cuda_str_version, torch.version.cuda))
      building 'diffoctreerast._C' extension
      creating build\temp.win-amd64-cpython-310
      creating build\temp.win-amd64-cpython-310\Release
      creating build\temp.win-amd64-cpython-310\Release\src
      creating build\temp.win-amd64-cpython-310\Release\src\octree_decoupoly_rasterizer
      creating build\temp.win-amd64-cpython-310\Release\src\octree_decoupoly_rasterizer\cuda
      creating build\temp.win-amd64-cpython-310\Release\src\octree_gaussian_rasterizer
      creating build\temp.win-amd64-cpython-310\Release\src\octree_gaussian_rasterizer\cuda
      creating build\temp.win-amd64-cpython-310\Release\src\octree_trivec_rasterizer
      creating build\temp.win-amd64-cpython-310\Release\src\octree_trivec_rasterizer\cuda
      creating build\temp.win-amd64-cpython-310\Release\src\octree_voxel_rasterizer
      creating build\temp.win-amd64-cpython-310\Release\src\octree_voxel_rasterizer\cuda
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\torch\csrc\api\include -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\TH -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\THC -IC:\pinokio\bin\miniconda\include -IC:\pinokio\api\TRELLIS.git\app\env\include -IC:\pinokio\bin\miniconda\include -IC:\pinokio\bin\miniconda\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" /EHsc /Tpsrc/ext.cpp /Fobuild\temp.win-amd64-cpython-310\Release\src/ext.obj /MD /wd4819 /wd4251 /wd4244 /wd4267 /wd4275 /wd4018 /wd4190 /wd4624 /wd4067 /wd4068 /EHsc -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 /std:c++17
      ext.cpp
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\torch\csrc\api\include -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\TH -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\THC -IC:\pinokio\bin\miniconda\include -IC:\pinokio\api\TRELLIS.git\app\env\include -IC:\pinokio\bin\miniconda\include -IC:\pinokio\bin\miniconda\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" /EHsc /Tpsrc/octree_decoupoly_rasterizer/api.cpp /Fobuild\temp.win-amd64-cpython-310\Release\src/octree_decoupoly_rasterizer/api.obj /MD /wd4819 /wd4251 /wd4244 /wd4267 /wd4275 /wd4018 /wd4190 /wd4624 /wd4067 /wd4068 /EHsc -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 /std:c++17
      api.cpp
      C:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\utils\cpp_extension.py:1964: UserWarning: TORCH_CUDA_ARCH_LIST is not set, all archs for visible cards are included for compilation.
      If this is not desired, please set os.environ['TORCH_CUDA_ARCH_LIST'].
        warnings.warn(
      C:\pinokio\bin\miniconda\bin\nvcc -c src/octree_decoupoly_rasterizer/cuda/backward.cu -o build\temp.win-amd64-cpython-310\Release\src/octree_decoupoly_rasterizer/cuda/backward.obj -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\torch\csrc\api\include -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\TH -IC:\pinokio\api\TRELLIS.git\app\env\lib\site-packages\torch\include\THC -IC:\pinokio\bin\miniconda\include -IC:\pinokio\api\TRELLIS.git\app\env\include -IC:\pinokio\bin\miniconda\include -IC:\pinokio\bin\miniconda\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" -Xcudafe --diag_suppress=dll_interface_conflict_dllexport_assumed -Xcudafe --diag_suppress=dll_interface_conflict_none_assumed -Xcudafe --diag_suppress=field_without_dll_interface -Xcudafe --diag_suppress=base_class_has_different_dll_interface -Xcompiler /EHsc -Xcompiler /wd4068 -Xcompiler /wd4067 -Xcompiler /wd4624 -Xcompiler /wd4190 -Xcompiler /wd4018 -Xcompiler /wd4275 -Xcompiler /wd4267 -Xcompiler /wd4244 -Xcompiler /wd4251 -Xcompiler /wd4819 -Xcompiler /MD -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -IC:\pinokio\cache\TMPDIR\pip-req-build-bp3x90gm\lib/glm/ -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -std=c++17 --use-local-env
      C:\pinokio\bin\miniconda\include\crt/host_config.h(153): fatal error C1189: #error:  -- unsupported Microsoft Visual Studio version! Only the versions between 2017 and 2022 (inclusive) are supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.
      backward.cu
      error: command 'C:\\pinokio\\bin\\miniconda\\bin\\nvcc.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> diffoctreerast

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

```

### Potential Explanation
- Slight mis-match with CUDA versions??

**Comment by Deathdadev** [comment]
### **Issue**: `gradio` Not Found After Successful Installation
The installation process appears to have been resolved with commit [342f9f9](https://github.com/Deathdadev/TRELLIS/commit/342f9f91cb522d775c822067fba8654eb277cdd1). However, when running the start script, an error occurs indicating that the gradio module is not installed, even though the installation logs show it was successfully installed.

### Install.js logs snippet
```
Successfully installed aiofiles-23.2.1 annotated-types-0.7.0 anyio-4.7.0 click-8.1.7 devicetorch-0.1.13 exceptiongroup-1.2.2 fastapi-0.115.6 ffmpy-0.4.0 gradio-4.44.1 gradio-client-1.3.0 gradio_litmodel3d-0.0.1 h11-0.14.0 httpcore-1.0.7 httpx-0.28.1 importlib-resources-6.4.5 jinja2-3.1.4 markdown-it-py-3.0.0 markupsafe-2.1.5 mdurl-0.1.2 orjson-3.10.12 pandas-2.2.3 pillow-10.4.0 pydantic-2.10.3 pydantic-core-2.27.1 pydub-0.25.1 pygments-2.18.0 python-multipart-0.0.19 pytz-2024.2 rich-13.9.4 ruff-0.8.2 semantic-version-2.10.0 shellingham-1.5.4 sniffio-1.3.1 starlette-0.41.3 tomlkit-0.12.0 typer-0.15.1 tzdata-2024.2 uvicorn-0.32.1 websockets-12.0
```

### Start.js logs
```
Microsoft Windows [Version 10.0.26100.2314]
(c) Microsoft Corporation. All rights reserved.

C:\pinokio\api\TRELLIS.git\app>conda_hook && conda deactivate && conda deactivate && conda deactivate && conda activate base && C:\pinokio\api\TRELLIS.git\app\env\Scripts\activate C:\pinokio\api\TRELLIS.git\app\env && python app.py

Traceback (most recent call last):
  File "C:\pinokio\api\TRELLIS.git\app\app.py", line 1, in <module>
    import gradio as gr
ModuleNotFoundError: No module named 'gradio'
(env) (base) C:\pinokio\api\TRELLIS.git\app>

Disconnected...
```
